actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -84,7 +84,15 @@ def tag_options(options, escape = true)
               value.each_pair do |k, v|
                 next if v.nil?
 
-                v = (v.is_a?(Array) || v.is_a?(Hash)) ? safe_join(TagHelper.build_tag_values(v), " ") : v.to_s
+                case v
+                when Array, Hash
+                  tokens = TagHelper.build_tag_values(v)
+                  next if tokens.none?
```

Comment:
Without this change, the desired outcome can be achieved with a combination of a `#token_list` and `#presence` call:

```html+erb
<% post = Post.new %>

<%= form_with model: post do |form| %>
  <%= form.text_field :title, aria: { describedby: token_list(post_title_error: post.errors[:title].any?).presence } %>
  <%= tag.span(post.errors[:title].to_sentence, id: :post_title_error) if post.errors[:title].any? %>
<% end %>
```